### DISCUSSION TITLE
Resetting a template before applying it

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3896] [1. Help] [answered]
I'm using the following code to insert an add every x row. In general it works, however, it seems to accumulate all the banners instead of showing just the one each row.

How can I reset/clear the template before it will be rendered?

**Code**
![image](https://github.com/alpinejs/alpine/assets/1435963/5c500e4f-2864-4f96-ae60-59529b8c164a)

**Output**
![image](https://github.com/alpinejs/alpine/assets/1435963/202ff3d7-280d-487a-8a4e-2b5540e02074)

**Comment by ekwoka** [comment]
I'm not sure what you mean. What does "reset the template" mean?

**Reply by simonferndriger** [reply]
With "resetting" the template, I mean that when the template is rendered, it takes what was initially in it (in the source code) and does not "save" any DOM changes after it was rendered. So it is reset or rendered anew every time the x-if condition is true. Is that even possible?

**Reply by SimoTod** [reply, marked as answer]
That should be the default behaviour. X-if clones the content of the template when the condition is evaluated to true and destroys the cloned element when thr condition is evaluated to false.

**Reply by ekwoka** [reply]
Yeah, that should be the standard behavior.

**Reply by simonferndriger** [reply]
Ah I see - thanks. In that case the issue seems to be with the third party script.
